### PR TITLE
Handle disposed TestRunRequest object in Abort.

### PR DESCRIPTION
**Pull request by Faizan2304**
Resolves: https://github.com/Microsoft/vstest/issues/1256

## Description
We were calling Abort and Cancel on disposed TestRunRequest Object

## Related issue
https://github.com/Microsoft/vstest/issues/1256

### CODE REVIEW

File: src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs [hunk 1]
```diff
@@ -236,7 +236,7 @@ public void CancelAsync()
             {
                 if (this.disposed)
                 {
-                    throw new ObjectDisposedException("testRunRequest");
+                    EqtTrace.Warning("Ignoring TestRunRequest.CancelAsync() as testRunRequest aboject has already been disposed.");
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs]: return here.

**Review by singhsarab — APPROVED** [review]
with suggestion

File: src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs (hunk 1 above)
Comment (Faizan2304) [reply src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs]: Fixed

File: src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs [hunk 2]
```diff
@@ -236,7 +236,8 @@ public void CancelAsync()
...
```
Comment (smadala) [inline src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs]: Nit: object

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 3]
```diff
@@ -476,31 +476,32 @@ private bool RunTests(IRequestData requestData, TestRunCriteria testRunCriteria,
...
-                            // Wait for the run completion event
-                            testRunRequest.WaitForCompletion();
-                        }
-                        finally
-                        {
-                            this.testLoggerManager.UnregisterTestRunEvents(testRunRequest);
-                            this.testRunResultAggregator.UnregisterTestRunEvents(testRunRequest);
-                            testRunEventsRegistrar?.UnregisterTestRunEvents(testRunRequest);
-                        }
+                        this.testLoggerManager.UnregisterTestRunEvents(this.currentTestRunRequest);
```
Comment (smadala) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: Unregistered testrunrequest getting used in below catch.

**Review by smadala — APPROVED** [review]

File: src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs (hunk 2 above)
Comment (Faizan2304) [reply src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs]: Fixed

**Pull request merged**